server/game_components/game_objects.py

```diff
@@ -93,80 +96,140 @@ def __init__(self, _name: str, _allowed_actions: list[str]):
         super().__init__(_name, _allowed_actions)
 
 
+class AddCommandResult(typing.NamedTuple):
+    success: bool
+    message: str
+
+
 class Player(Entity):
     level: int
-    command_queue: list[dict[str, str | int | None]]
+    command_queue: list[dict[str, Entity | None]]
 
-    def __init__(self, _name: str, _allowed_actions: list[str]):
+    def __init__(self, _name: str, _allowed_actions: list[str], game: Game):
         super().__init__(_name, _allowed_actions)
         self.level = 0
         self.command_queue = []
+        self.game = game
 
-    def update(self) -> list[dict[str, str | int | None]] | None:
-        directions = ["north", "east", "south", "west"]
+    def update(self) -> list[ActionDict] | MovementDict | None:
+        """Updates the player for one tick.
+
+        Executes the next action in the players queue.
+        Returns a list of ActionDicts if the player did something and None if they didn't
+        """
         self.mana += 10
         self.health += random.randint(1, 5)
 
-        commands: list[dict[str, str | int | None]] = []
-        while len(self.command_queue) > 0:
+        result = None
+
+        if len(self.command_queue) > 0:
             next_command = self.command_queue.pop()
-            if self.in_combat:
-                if next_command["command"] not in directions:
-                    commands.append(next_command)
-            if not self.in_combat:
-                # this is wrong too
-                pass
-            if next_command["command"] not in directions:
-                break
+            result = self._do(next_command)
 
         super().update()
-        if len(commands) > 0:
-            return commands
-        return None
+
+        return result
+
+    def _do(self, command: CommandDict) -> list[ActionDict] | MovementDict | None:
+        movement_commands = ["north", "east", "south", "west"]
+
+        result = None
+        # TODO: Tell the player "HEY, YOU DID THIS THING"
+        if command["command"] in movement_commands:
+            valid_move = self.game.move_player(self, command["command"])
+            result = {
+                "player": self.uid,
+                "direction": command["command"],
+                "success": valid_move,
+            }
+        elif command["command"] == "flee":
+            pass  # TODO: IMPLEMENT FLEEING.
```

Comment:
What does fleeing mean? I'm not sure we have a game mechanic like this. (I don't think we get locked into battle like an RPG, unless we do?)